lib/util.js

```diff
@@ -748,28 +748,44 @@ var util = {
   /**
    * @api private
    */
-  addPromisesToRequests: function addPromisesToRequests(constructor, PromiseDependency) {
-    PromiseDependency = PromiseDependency || null;
-    if (!PromiseDependency && typeof Promise !== 'undefined') {
+  addPromises: function addPromises(constructors, PromiseDependency) {
+    if (PromiseDependency === undefined && AWS && AWS.config) {
+      PromiseDependency = AWS.config.getPromisesDependency();
+    }
+    if (PromiseDependency === undefined && typeof Promise !== 'undefined') {
       PromiseDependency = Promise;
     }
-    if (typeof PromiseDependency !== 'function') {
-      delete constructor.prototype.promise;
-      return;
+    if (typeof PromiseDependency !== 'function') var deletePromises = true;
+    if (!Array.isArray(constructors)) constructors = [constructors];
+
+    for (var ind = 0; ind < constructors.length; ind++) {
+      var classPrototype = constructors[ind].prototype;
+      if (deletePromises) {
+        if (classPrototype.deletePromisesFromClass) {
+          classPrototype.deletePromisesFromClass();
```

Comment:
This works, and is very close to what I was trying to describe before. Is there a reason you put the `deletePromisesFromClass` and `addPromisesToClass` on the prototype instead of on the constructor? This works, but the code within those methods is a little confusing, since the `this` context won't be an instance of the method as you'd typically expect, it would be the prototype itself.